### DISCUSSION TITLE
How to compile the HelloWorld efi module on CentOS Linux 7 (Core)?

### DISCUSSION THREAD

**Discussion by sayeesh** [tianocore/edk2#3902] [Q&A]
How to compile the HelloWorld efi module on CentOS Linux 7 (Core)?

I did below but its not working out well.

git clone https://github.com/tianocore/edk2.git
then from edk2 folder ran below commands
source edksetup.sh
build -p MdeModulePkg/MdeModulePkg.dsc -m MdeModulePkg/Application/HelloWorld/HelloWorld.inf -a IA32 -b DEBUG -t GCC48

am getting below error

[samadall@bgl-vms-vm1224 edk2]$ build -p MdeModulePkg/MdeModulePkg.dsc -m MdeModulePkg/Application/HelloWorld/HelloWorld.inf -a IA32 -b D
EBUG -t GCC48
Build environment: Linux-3.10.0-1160.42.2.el7.x86_64-x86_64-with-centos-7.9.2009-Core
Build start time: 09:41:23, Jan.16 2023

WORKSPACE        = /home/samadall/M6_PROJ/Hello_world/edk2
EDK_TOOLS_PATH   = /home/samadall/M6_PROJ/Hello_world/edk2/BaseTools
CONF_PATH        = /home/samadall/M6_PROJ/Hello_world/edk2/Conf
PYTHON_COMMAND   = /usr/bin/python3.6

Processing meta-data
.
Architecture(s)  = IA32
Build target     = DEBUG
Toolchain        = GCC48

Active Platform          = /home/samadall/M6_PROJ/Hello_world/edk2/MdeModulePkg/MdeModulePkg.dsc
Active Module            = /home/samadall/M6_PROJ/Hello_world/edk2/MdeModulePkg/Application/HelloWorld/HelloWorld.inf
.

build.py...
/home/samadall/M6_PROJ/Hello_world/edk2/MdeModulePkg/MdeModulePkg.dec(28): error 000E: File/directory not found in workspace
        /home/samadall/M6_PROJ/Hello_world/edk2/MdeModulePkg/Library/BrotliCustomDecompressLib/brotli/c/include

- Failed -
Build end time: 09:41:25, Jan.16 2023
Build total time: 00:00:02

Is there something i am missing?

**Comment by tormodvolden** [comment]
I am late to the party, but yes, you must run `git submodule update --init` after cloning.

https://github.com/tianocore/tianocore.github.io/wiki/Common-instructions
